src/v2/models/AppState.js

```diff
@@ -106,6 +106,28 @@ export default Model.extend({
     return currentViewState;
   },
 
+  /**
+   * Checks to see if we're in an enrollment or challenge flow.
+   * @returns {boolean}
+   */
+  isAuthenticatorChallenge () {
+    const currentAuthenticator = this.get('currentAuthenticator');
+    const currentAuthenticatorEnrollment = this.get('currentAuthenticatorEnrollment');
+    const authenticatorEnrollments = this.get('authenticatorEnrollments')?.value || [];
+
+    // For single-enrollment (currentAuthenticatorEnrollment)
+    if (currentAuthenticatorEnrollment && currentAuthenticator.id === currentAuthenticatorEnrollment.id) {
```

Comment:
Are `currentAuthenticator` and `currentAuthenticatorEnrollment` both returned for all responses? I also see some mock responses returning both `currentAuthenticatorEnrollment` and `authenticatorEnrollments`.. It is possible our mocks are not accurate though. cc @SamSanjabi-okta 

What about checking the form name instead to figure out if we are in challenge flow?

Reply:
I don't think `currentAuthenticator` and `currentAuthenticatorEnrollment` should ever _both_ be in the response.  In general:
- `currentAuthenticator` = current authenticator being challenged during enrollment, or during verification for authenticators verified by type (like webauthn) -- right now RSA & Symantec might also be doing they shouldn't and it's something we'll fix later.
- `currentAuthenticatorEnrollment` = current authenticator being challenged during verification for all other cases (single enrollment authenticators, multiple enrollment authenticators when a device is selected like phone, etc...)

`authenticatorEnrollments` I believe comes back in all responses after the user is identified and has all of user's enrollments, unless it's the "anonymized" password + email screen in which case it just has those two but the object will still be there.

Reply:
From what I have seen, `currentAuthenticator` is always present for enroll or challenge flows. The `currentAuthenticatorEnrollment` may or may not be present, hence the conditional check here for its existence.

We could check the form name, but I figured a more future-proof solution was to always check if the `currentAuthenticator` exists in either the `currentAuthenticatorEnrollment` or `authenticatorEnrollments` fields.

Reply:
> currentAuthenticator is always present for enroll or challenge flows

What Sam said above is that it depends on the authenticator. You are probably seeing it for symantec because on the backend, it is verified by type (like webauthn). My understanding is that we will see `currentAuthenticator` for authenticators with type `app` and ones with type `security_key`. Are you seeing this for other authenticator types too? 

From the info in this thread, it doesn't seem like we can use these objects 

- `authenticatorEnrollments` is always returned after the user is identified - this means after identify, in either enroll or verify flows
- `currentAuthenticator` is always returned during enrollment. And only during verification of authenticators verified by type. It is not consistently there during the verify flow for all authenticators
- `currentAuthenticatorEnrollment` is always returned on verify but ONLY for those authenticators NOT verified by type.

I think the best way to check if we are in challenge flow, at this time, is by form name.

Reply:
Given that we `currentAuthenticator` and `currentAuthenticatorEnrollment` should _ideally_ not be present, I'll update this logic to rely on the formName